### PR TITLE
Add deviceProfiles query

### PR DESCRIPTION
**Pull request by shanewilson**
**What kind of change does this PR introduce?**
Only being able to list deviceModels is boring. This adds deviceProfiles under PartnerProfile.

**Does this PR introduce a breaking change?**
No just additive

### CODE REVIEW

**Comment by shanewilson** [comment]
I don't understand the netlify error

**Comment by chuckcarpenter** [comment]
adding @tazsingh to reviewers, as he felt pretty strongly about keeping it boring since we don't have the UI to need device profile.

File: src/services/Device/schema.graphql [hunk 1]
```diff
@@ -3,18 +3,29 @@ interface Node {
...
 extend type Query {
-  deviceModels(nameFilter: String, after: ID, before: ID, first: Int, last: Int): DeviceModelRelayConnection!
+  deviceModels(after: ID, before: ID, first: Int, last: Int): DeviceModelRelayConnection!
+  deviceProfiles(after: ID, before: ID, first: Int, last: Int): DeviceProfileRelayConnection!
 }
 
 type DeviceProfile implements Node @key(fields: "id") {
   id: ID!
-  deviceIdentifier: String!
+  deviceIdentifier: ID!
```
Comment (chuckcarpenter) [inline src/services/Device/schema.graphql]: _should_ be unique, so why not

**Comment by tazsingh** [comment]
@chuckcarpenter @shanewilson Yeah the consideration from my side was to add the field for device profiles where they'll be accessed from. That is, if they're accessed from a partner profile, the field should be there, or if it's for a given user account, the field should be there, and so on. I don't believe it's a top-level `Query` field as device profiles are always related to something else.

I see that we didn't add a field to actually retrieve the device profiles however... I'm assuming that it should be added to the `PartnerProfile` at the least. Should it be added to `UserAccount` as well? Anywhere else?

**Comment by shanewilson** [comment]
this is currently a query that powers the dashboard:
https://github.com/resideo/pro-portal/blob/master/packages/web/src/components/dashboard/deviceActivationList.tsx#L149

I'm not sure what the filter on that would be - I guess devices the logged in user can see?

**Comment by tazsingh** [comment]
If I'm reading that query correctly (and I very much could not be), it appears to be all devices for a `PartnerProfile`. Therefore we should add a `devices` field to the `PartnerProfile` type?

What are the various states for devices? `ACTIVATED`/`PENDING`?/`INACTIVE`? That should probably be an enum on the connection to allow for filtering of that `devices` list.

**Pull request merged**